src/actions/appSettingsActions.js

```diff
@@ -179,3 +180,24 @@ export const setUserJoinedBetaAction = (userJoinedBeta: boolean, ignoreSuccessTo
     });
   };
 };
+
+
+export const changeAppThemeAction = () => {
+  return (dispatch: Dispatch, getState: GetState) => {
+    const {
+      appSettings: { data: { theme: { mode } } },
+    } = getState();
+
+    const newMode = mode === LIGHT_MODE ? DARK_MODE : LIGHT_MODE;
+    const colors = mode === LIGHT_MODE ? darkThemeColors : lightThemeColors;
+
+    const theme = { mode: newMode, colors };
+    dispatch(saveDbAction('app_settings', { appSettings: { theme } }));
+    dispatch({
+      type: UPDATE_APP_SETTINGS,
+      payload: {
+        theme,
+      },
+    });
```

Comment:
action dispatch in the line above takes almost the same space of code, maybe this can be done in one liner as well?

Reply:
I don't think we need to squeeze things on one line just because. The important thing is not just the length, but the complexity, and the line above seems a bit simpler, in my opinion. Maybe we could have a function instead that we can reuse on the rest of the actions:

```js
const updateSettingsAction = (payload: Object) => ({
  type: UPDATE_APP_SETTINGS,
  payload,
});

// then...

dispatch(updateSettingsAction({ theme }));
```

Anyways, we don't need to add extra stuff on this PR, we can do that later and leave this as is. What do you think?

Reply:
Hm, maybe it's just me, but I'm personally more used to dispatches that list type and payload in separate lines. In sake of fast readability. (And we usually use the same approach / way across this app).
I've put theme in one line to make it shorter :)

```
dispatch({
   type: UPDATE_APP_SETTINGS,
   payload: { theme },
});
```

Not sure about additional function, though. Isn't basic dispatch more understandable?

Reply:
Well we could discuss if we prefer one way or another. I don't think it's really a problem anyway, but IMHO, but there could be a couple of advantages on using a function for that.

The first one is removing some redundant stuff that gets repeated everywhere, like `type:` and `payload:`, this results on less noisy code, which is easier to write and read.

In other hand, it detaches the action structure from the callers. This helps avoid common errors like typos or missing/extra information. This also helps us write the code in a way that's easier for linters to detect those errors before we even have to test the code.
And of course, if we ever want to change that structure, it would be way easier, and with the help of the linter, any outdated call should be detected right away.

Another less probable scenario would be if we want to convert that action into a thunk action (or a saga, probably, when we get there 😄). In that case we would only need to change the function and that's it.

As I said before, I don't like the sole motivation of having fewer lines. I think we should focus on reducing the complexity, instead. This is not a particularly complex portion of code, so I wouldn't worry too much about it right now.